darkside-tests/tests/tests.rs

```diff
@@ -245,92 +246,124 @@ async fn sent_transaction_reorged_into_mempool() {
     let secondary = environment.create_client().await;
     primary.do_sync(false).await.unwrap();
 
-    let proposal =
-        to_clients_proposal(&primary, &[(&secondary, Shielded(Orchard), 100_000, None)]).await;
-
-    let mut send_height = 0;
+    let proposal = testutils::chain_generics::with_assertions::to_clients_proposal(
+        &primary,
+        &[(&secondary, Shielded(Orchard), 100_000, None)],
+    )
+    .await;
+    println!("following proposal, preparing to unwind if an assertion fails.");
 
-    let txids = &primary
+    let txids = primary
         .complete_and_broadcast_stored_proposal()
         .await
         .unwrap();
 
-    println!(
-        "{:?}",
-        zingolib::testutils::lightclient::list_txids(&primary).await
-    );
-
-    let _recorded_fee = *zingolib::testutils::assertions::lookup_fees_with_proposal_check(
-        &primary, &proposal, txids,
-    )
-    .await
-    .first()
-    .expect("one transaction must have been proposed")
-    .as_ref()
-    .expect("record must exist");
-
-    zingolib::testutils::lightclient::lookup_statuses(&primary, txids.clone())
-        .await
-        .map(|status| {
-            assert_eq!(
-                status,
-                Some(ConfirmationStatus::Transmitted(send_height.into()))
-            );
-        });
-
-    zingolib::testutils::lightclient::lookup_statuses(&secondary, txids.clone())
-        .await
-        .map(|status| {
-            assert!(status.is_none());
-        });
+    // a modified zingolib::testutils::chain_generics::with_assertions::follow_proposal block
+    {
+        let sender = &primary;
+        let proposal = &proposal;
 
-    environment
-        .get_connector()
-        .clear_incoming_transactions()
+        let server_height_at_send = BlockHeight::from(
+            grpc_connector::get_latest_block(environment.lightserver_uri().unwrap())
+                .await
+                .unwrap()
+                .height as u32,
+        );
+
+        // check that each record has the expected fee and status, returning the fee
+        let (sender_recorded_fees, (sender_recorded_outputs, sender_recorded_statuses)): (
+            Vec<u64>,
+            (Vec<u64>, Vec<ConfirmationStatus>),
+        ) = testutils::assertions::for_each_proposed_record(
+            sender,
+            proposal,
+            &txids,
+            |records, record, step| {
+                (
+                    testutils::assertions::compare_fee(records, record, step),
+                    (record.query_sum_value(OutputQuery::any()), record.status),
+                )
+            },
+        )
         .await
-        .unwrap();
-    environment.bump_chain().await;
+        .into_iter()
+        .map(|stepwise_result| {
+            stepwise_result
+                .map(|(fee_comparison_result, others)| (fee_comparison_result.unwrap(), others))
+                .unwrap()
+        })
+        .unzip();
```

Comment:
this could be refactored to be a lot more readable. the idea here is very simple but there is a lot of mapping and nested results that is cumbersome to follow. non-blocking